sparse/tests/test_core.py

```diff
@@ -130,12 +131,43 @@ def test_dot():
     a = random_x((3, 4, 5))
     b = random_x((5, 6))
 
+    la = a.tolist()
+    lb = b.tolist()
+    la, lb          # silencing flake8
+
     sa = COO.from_numpy(a)
     sb = COO.from_numpy(b)
 
     assert_eq(a.dot(b), sa.dot(sb))
     assert_eq(np.dot(a, b), sparse.dot(sa, sb))
 
+    if sys.version_info >= (3, 5):
+        # Basic equivalences
+        assert_eq(eval("a @ b"), eval("sa @ sb"))
+        assert_eq(eval("sa @ sb"), sparse.dot(sa, sb))
+
+        # Test that SOO's and np.array's combine correctly
+        assert_eq(eval("a @ sb"), eval("sa @ b"))
```

Comment:
Weird.. that error comes from the fact that ``np.dot(sa, sb)`` ends up calling ``sa.__mul__(sb)``, which... I have no idea why it would do that.